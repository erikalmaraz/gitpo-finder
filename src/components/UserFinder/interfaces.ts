interface Props {
  placeholder?: string;
}

export default Props;
