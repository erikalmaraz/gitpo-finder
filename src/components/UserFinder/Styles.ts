import styled from "styled-components";

export const Finder = styled.input`
  width: 100%;
  max-width: 600px;
  border-radius: 6px;
  border: 1px solid #999;
  padding: 10px;
  margin: 10px 0;
  background: #ddd;
`;

export const SubmitButton = styled.button.attrs({
  type: "submit",
})`
  background: #333;
  color: #fff;
  text-align: center;
  border-radius: 5px;
  border: 0;
  padding: 8px;
  margin-top: 10px;
  font-weight: bold;
  min-width: 120px;
  font-size: 16px;
`;
