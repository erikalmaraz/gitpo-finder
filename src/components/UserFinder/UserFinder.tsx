import React, { FC, useState } from "react";
import { Finder, SubmitButton } from "./Styles";
import props from "./interfaces";
import { navigate } from "@reach/router";

const UserFinder: FC<props> = ({ placeholder }: props) => {
  const [username, setUsername] = useState("");
  const searchByUser = () => {
    navigate(`/user/${username}`);
  };

  return (
    <>
      <div>
        <Finder
          onChange={(e) => {
            setUsername(e.target.value);
          }}
        />
      </div>
      <div>
        <SubmitButton onClick={() => searchByUser()}>Go!</SubmitButton>
      </div>
    </>
  );
};

export default UserFinder;
