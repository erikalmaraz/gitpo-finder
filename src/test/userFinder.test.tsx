import React from "react";
import { render, screen } from "@testing-library/react";
import UserFinder from "../components/UserFinder/UserFinder";
import renderer from "react-test-renderer";

test("Should render UserFinder correctly", () => {
  const tree = renderer.create(<UserFinder />).toJSON();
  expect(tree).toMatchSnapshot();
});

