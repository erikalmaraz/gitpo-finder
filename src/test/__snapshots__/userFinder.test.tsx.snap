// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render UserFinder correctly 1`] = `
Array [
  <div>
    <input
      className="sc-bdfBQB kVKBGV"
      onChange={[Function]}
      placeholder="Type a username..."
      type="text"
    />
  </div>,
  <div>
    <button
      className="sc-gsTEea ckPYpT"
      onClick={[Function]}
      type="submit"
    >
      Go!
    </button>
  </div>,
]
`;
